### PR TITLE
Add io.github.equicord.equibop

### PR DESCRIPTION
**Pull request by verysillycat**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
Equibop is a fork of Vesktop which aims at adding extra plugins, but still keeping Discord optimized and running smoothly with ease
The Creator of [Sunroof](https://github.com/verticalsync/Sunroof) with flathub package [io.github.verticalsync.sunroof](https://github.com/flathub/io.github.verticalsync.sunroof) announced that it reached EOL and requested to people to [migrate to Equibop for better maintaining and more](https://github.com/verticalsync/Sunroof?tab=readme-ov-file#eol-read-thanks), Prompting the Request of adding this package.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: startequibop [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    if [[ -e "$WAYLAND_SOCKET" ]]; then
+        echo "Wayland socket is available, running natively on Wayland."
+        echo "To disable, remove the --socket=wayland permission."
+        FLAGS+=(--ozone-platform-hint=auto)
+    fi
+fi
+
+echo "Passing the following arguments to Electron:" "${FLAGS[@]}"
+zypak-wrapper /app/bin/equibop/equibop.bin "${FLAGS[@]}" "$@"
```
Comment (Lawstorant) [inline startequibop:26]: This whole file should be upstream. Flatpak repo is not a place for source/binaries

**Review by Lawstorant — CHANGES_REQUESTED** [review]

File: io.github.equicord.equibop.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.kde.StatusNotifierWatcher # Tray functionalities on KDE
+  - --device=all # Needed for GPU acceleration and the webcam
+  - --filesystem=xdg-videos:ro
+  - --filesystem=xdg-pictures:ro
+  - --filesystem=xdg-download # This and the above two are used for drag-n-drop, and download managing
+  - --filesystem=xdg-run/pipewire-0 # Pipewire interfacing
+  - --filesystem=xdg-run/speech-dispatcher # For TTS and VcNarrator
```
Comment (bbhtt) [inline io.github.equicord.equibop.yml:25]:
```suggestion
  - --filesystem=xdg-run/speech-dispatcher:ro # For TTS and VcNarrator
```

File: startequibop (hunk 1 above)
Comment (bbhtt) [reply startequibop:26]: Not it is fine `zypak-wrapper` is not used anywhere else other than Flatpak.

Comment (bbhtt) [reply startequibop:26]: Not it is fine `zypak-wrapper` is not used anywhere else other than Flatpak.

**Comment by bbhtt** [comment]
> The Creator of [Sunroof](https://github.com/verticalsync/Sunroof) with flathub package [io.github.verticalsync.sunroof](https://github.com/flathub/io.github.verticalsync.sunroof) announced that it reached EOL and

In this case it is better to wait to see if the project is alive, than have this EOL-ed again after 2 months of submission.

**Comment by verysillycat** [comment]
> > The Creator of [Sunroof](https://github.com/verticalsync/Sunroof) with flathub package [io.github.verticalsync.sunroof](https://github.com/flathub/io.github.verticalsync.sunroof) announced that it reached EOL and
> 
> In this case it is better to wait to see if the project stays alive and maintained, than have this EOL-ed again after 2 months of submission.

I believe it will since it just doesn't have one single maintainer and instead is a team (including the creator of Sunroof) behind it

File: io.github.equicord.equibop.yml (hunk 2 above)
Comment (verysillycat) [reply io.github.equicord.equibop.yml]: Accepted. [`4012b115a7e8ea27093c62d2ec2590ab555ef180`](https://github.com/flathub/flathub/pull/5464/commits/4012b115a7e8ea27093c62d2ec2590ab555ef180)

**Comment by bbhtt** [comment]
bot, build io.github.equicord.equibop

**Comment by bbhtt** [comment]
release dates aren't ordered properly in metainfo.

**Comment by verysillycat** [comment]
> release dates aren't ordered properly in metainfo.

this was fixed on [da82ca1](https://github.com/flathub/flathub/pull/5464/commits/da82ca1497725878b2bcccc6a7a8f882161a04a6) may you please try again?

**Comment by bbhtt** [comment]
You can start the build

bot, build io.github.equicord.equibop

**Comment by verysillycat** [comment]
> You can start the build
> 
> bot, build io.github.equicord.equibop

Oh I see thanks however is "Commiting builds" supposed to take this long?
image

**Comment by verysillycat** [comment]
Nevermind

**Comment by bbhtt** [comment]
Doesn't start

image

**Comment by verysillycat** [comment]
bot, build io.github.equicord.equibop

**Comment by verysillycat** [comment]
bot, build io.github.equicord.equibop

**Comment by verysillycat** [comment]
Okay, I'm now fully confident that there are no more issues, since it builded correctly in flathub and i tested it locally and it worked just fine.

**Comment by sdhEmily** [comment]
bot, build io.github.equicord.equibop

**Comment by verysillycat** [comment]
> I see a lot of errors when launching, is that expected ?
> 
> ```
> (node:3) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/bbhtt/.var/app/io.github.equicord.equibop/config/equibop/settings/quickCss.css'
>     at async open (node:internal/fs/promises:639:25)
> (Use `equibop.bin --trace-warnings ...` to show where the warning was created)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 5)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 6)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 7)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 8)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 9)
> { discordBranch: 'stable' }
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 10)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 11)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 12)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 13)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 14)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 16)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 17)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 18)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 22)
> (node:3) UnhandledPromiseRejectionWarning: Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.
>     at node:electron/js2c/sandbox_bundle:2:58441
>     at IpcRendererInternal.<anonymous> (node:electron/js2c/sandbox_bundle:2:52634)
>     at IpcRendererInternal.emit (node:electron/js2c/sandbox_bundle:2:34768)
>     at Object.onMessage (node:electron/js2c/sandbox_bundle:2:50712)
> (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 23)
> ```
The first error seems to happen because you don't have anything added to quickCSS therefore the file doesn't get created so it doesn't exist, you can safely ignore it, and about the other ones I'm not sure what could have do it, but I'm pretty sure it doesn't affect anything also because this related to the electron package itself

**Pull request closed without merging**